lib/upload.js

```diff
@@ -510,14 +513,27 @@ class Upload {
     if (end === Infinity || end > this._size) {
       end = this._size;
     }
-
+    
     this._source.slice(start, end, (error, chunk) => {
       if (error) {
         this._emitError(new DetailedError(`tus: could not slice file or stream (from ${start} to ${end})`, error));
         return;
       }
 
-      xhr.send(chunk);
+      // In Apache Cordova applications, a FileEntry must be resolved using
+      // FileReader instances, see
+      // https://cordova.apache.org/docs/en/8.x/reference/cordova-plugin-file/index.html#read-a-file
+      if (isCordova()) {
+        readAsByteArray(chunk, (error, byteArray) => {
+          if (error) {
+              this._emitError(new DetailedError(`tus: could not read file (from ${start} to ${end})`, error));
```

Comment:
Should we still `xhr.send` in the face of this error or is there a missing return?

Reply:
Moved that block to source.js in commit https://github.com/tus/tus-js-client/pull/134/commits/7380fdbbdf0883f8a6dbb6cc8f0aaf5346024d40. Could you confirm that is correct.

Reply:
It looks OK from what I can tell when looking at it from the git diff.